File: easy/1346_check_if_n_and_its_double_exist/solution.cpp
#include <unordered_set>
#include <vector>
class Solution {
public:
  bool checkIfExist(std::vector<int> &arr) {
    std::unordered_set<int> numsSet;

    for (const int &num : arr) {
      if (numsSet.find(num * 2) != numsSet.end() ||
          (num % 2 == 0 && numsSet.find(num / 2) != numsSet.end()))
        return true;

      numsSet.insert(num);
    }

    return false;
  }
};

File: easy/1346_check_if_n_and_its_double_exist/solution.py
class Solution:
    def checkIfExist(self, arr: list[int]) -> bool:
        numsSet = set()

        for num in arr:
            if num * 2 in numsSet or (num % 2 == 0 and num / 2 in numsSet):
                return True

            numsSet.add(num)

        return False
